worlds/yugioh06/__init__.py

```diff
@@ -0,0 +1,538 @@
+
+                else:
+                    entrance.access_rule = \
+                        (lambda opp, item, amount: lambda state: state.has_group(item, self.player, amount) and
+                                                                 state.yugioh06_difficulty(self.player, opp.difficulty)
+                                                                 and state.has_all(opp.additional_info, self.player))\
+                                                                 (opponent, unlock_item, unlock_amount)
+            else:
+                entrance.access_rule = (lambda unlock, opp: lambda state:
+                state.has(unlock, self.player) and state.yugioh06_difficulty(self.player, opp.difficulty))(unlock_item, opponent)
+            campaign.exits.append(entrance)
+            entrance.connect(region)
+            self.multiworld.regions.append(region)
+
+        card_shop = self.multiworld.get_region('Card Shop', self.player)
+        # Booster Contents
+        for booster in booster_packs:
+            region = create_region(self,
+                                   booster, get_booster_locations(booster))
+            entrance = Entrance(self.player, booster, card_shop)
+            entrance.access_rule = (lambda unlock: lambda state: state.has(unlock, self.player))(booster)
+            card_shop.exits.append(entrance)
+            entrance.connect(region)
+            self.multiworld.regions.append(region)
+
+        challenge_region = self.multiworld.get_region('Challenges', self.player)
+        # Challenges
+        for challenge, lid in (Limited_Duels | Theme_Duels).items():
+            if challenge in self.removed_challenges:
+                continue
+            region = create_region(self,
+                                   challenge, {challenge: lid, challenge + " Complete": None})
+            entrance = Entrance(self.player, challenge, challenge_region)
+            entrance.access_rule = (lambda unlock: lambda state: state.has(unlock + " Unlock", self.player))(challenge)
+            challenge_region.exits.append(entrance)
+            entrance.connect(region)
+            self.multiworld.regions.append(region)
+
+    def generate_early(self):
+        if self.options.structure_deck.current_key == "none":
+            boosters = draft_boosters
+            if self.options.campaign_opponents_shuffle.value:
+                opponents = tier_1_opponents
+            else:
+                opponents = draft_opponents
+        else:
+            boosters = booster_packs
+            opponents = tier_1_opponents
+
+        if self.options.structure_deck.current_key == "random_deck":
+            self.options.structure_deck.value = self.random.choice([0, 1, 2, 3, 4, 5])
+        for item in self.options.start_inventory:
+            if item in opponents:
+                self.starting_opponent = item
+            if item in boosters:
+                self.starting_booster = item
```

Comment:
this for loop ignores the first booster and opponent if multiple are in start_inventory, right? is that intentional?

Reply:
It only matters if there is one in the start inventory or not.